src/client/driver/driver.js

```diff
@@ -339,18 +414,35 @@ export default class Driver {
         // but it should wait until the page is loaded before executing a command.
         this.readyPromise
             .then(() => {
+                // TODO: this.dialogMonitor check will be removed when we add native dialogs support for iframes
+                if (this.dialogsMonitor && command.expectedDialogs)
+                    this.dialogsMonitor.expectedDialogs = command.expectedDialogs;
+
+                var isCommandRejectableByError = isCommandRejectableByPageError(command);
+                var pendingPageError           = this.contextStorage.getItem(PENDING_PAGE_ERROR);
+
+                if (pendingPageError && isCommandRejectableByError) {
+                    this._onReady(new DriverStatus({ isCommandResult: true }));
+                    return;
+                }
+
+                var isThereActiveIframe = this.activeChildDriverLink ||
+                                          this.contextStorage.getItem(ACTIVE_IFRAME_SELECTOR);
+
+                if (!isExecutableInTopWindowOnly(command) && isThereActiveIframe) {
```

Comment:
Can we refactor this piece of code to the function with the meaningful name? This functions is already huge.